packages/cli/lib/utils.js

```diff
@@ -23,6 +23,8 @@ const Conflicter = require('yeoman-generator/lib/util/conflicter');
 
 const readdirAsync = promisify(fs.readdir);
 
+const RESERVED_PROPERTY_NAMES = ['constructor'];
```

Comment:
Let's also include access level keywords (like public) and any other applicable keywords from this list

Reply:
`public` is actually allowed as a property on a class ... surprising but TS didn't complain when I tried it.

Reply:
🤔